pkg/vmprovider/providers/vsphere2/virtualmachine/backup.go

```diff
@@ -28,128 +40,149 @@ type VMDiskData struct {
 
 // BackupVirtualMachine backs up the required data of a VM into its ExtraConfig.
 // Currently, the following data is backed up:
-// - Kubernetes VirtualMachine object in YAML format (without its .status field).
-// - VM bootstrap data in JSON (if provided).
-// - VM disk data in JSON (if created and attached by PVCs).
-func BackupVirtualMachine(ctx context.BackupVirtualMachineContextA2) error {
+// - VM relevant Kubernetes objects in YAML separated by "---".
+// - Cloud-Init instance ID (if not already stored in ExtraConfig).
+// - PVC disk data in JSON format (if DiskUUIDToPVC is not empty).
+func BackupVirtualMachine(opts BackupVirtualMachineOptions) error {
 	var moVM mo.VirtualMachine
-	if err := ctx.VcVM.Properties(ctx.VMCtx, ctx.VcVM.Reference(),
+	if err := opts.VcVM.Properties(opts.VMCtx, opts.VcVM.Reference(),
 		[]string{"config.extraConfig"}, &moVM); err != nil {
-		ctx.VMCtx.Logger.Error(err, "Failed to get VM properties for backup")
+		opts.VMCtx.Logger.Error(err, "Failed to get VM properties for backup")
```

Comment:
```suggestion
		opts.VMCtx.Logger.Error(err, "failed to get VM properties for backup")
```

Reply:
nit: Same for most errors in this file

Reply:
Updated all error logs accordingly in this PR.